packages/components/src/components/tooltip/tooltip.ts

```diff
@@ -256,40 +253,94 @@ export default class SdTooltip extends SolidElement {
   }
 
   render() {
+    const isStart = this.placement.endsWith('-start');
+    const isEnd = this.placement.endsWith('-end');
+
+    const skiddingMap = new Map([
+      ['start', '2'],
+      ['end', '-2']
+    ]);
+
+    const isSm = this.size === 'sm';
     return html`
       <sd-popup
         part="base"
+        id="tooltip"
         exportparts="
           popup:base__popup,
           arrow:base__arrow
         "
-        class=${classMap({
-      tooltip: true,
-      'tooltip--open': this.open
-    })}
+        class=${cx('tooltip', this.open && 'tooltip--open')}
         placement=${this.placement}
-        distance=${this.distance}
-        skidding=${this.skidding}
+        distance="12"
+        skidding=${((skiddingMap.get(isStart ? 'start' : isEnd ? 'end' : 'default') || 0) as number) * (isSm ? -1 : 1)}
         strategy=${this.hoist ? 'fixed' : 'absolute'}
         flip
         shift
         arrow
+        arrow-padding="0"
+        auto-size="vertical"
       >
-        <slot slot="anchor" aria-describedby="tooltip"></slot>
+        <slot slot="anchor" class=${cx(this.size === 'lg' ? 'text-xl' : 'text-base')}>
+          <sd-icon library="system" name="info-sign" class="sd-interactive rounded-full" tabindex="0"></sd-icon>
+        </slot>
 
         <slot
           name="content"
           part="body"
-          id="tooltip"
-          class="tooltip__body"
-          role="tooltip"
+          id="tooltip-body"
+          class="tooltip__body bg-primary text-white py-3 px-4 block shadow rounded-none text-sm text-left"
+          role="tooltiprole"
           aria-live=${this.open ? 'polite' : 'off'}
+          tabindex="0"
         >
           ${this.content}
         </slot>
       </sd-popup>
     `;
   }
+  static styles = [
+    SolidElement.styles,
+    unsafeCSS(InteractiveStyles),
+    css`
+      :host {
+        --hide-delay: 0ms;
+        --show-delay: 150ms;
+        display: contents;
+      }
+
+      #tooltip::part(popup) {
+        pointer-events: none;
+        z-index: 10;
+      }
+
+      #tooltip[placement^='top']::part(popup) {
+        transform-origin: bottom;
+      }
+
+      #tooltip[placement^='bottom']::part(popup) {
+        transform-origin: top;
+      }
+
+      #tooltip[placement^='left']::part(popup) {
+        transform-origin: right;
+      }
+
+      #tooltip[placement^='right']::part(popup) {
+        transform-origin: left;
+      }
+
+      #tooltip-body {
+        max-width: var(--max-width);
+      }
+
+      :host(:not([no-auto-size])) ::slotted(*:not([slot='anchor'])) {
+        overflow: auto;
+        max-width: var(--auto-size-available-width) !important;
+        max-height: var(--auto-size-available-height) !important;
+      }
```

Comment:
Is this still needed? I think you can savely remove `no-auto-size` and even the `:not([slot='anchor']` as we don't have such a slot here.